src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -69,7 +69,7 @@ internal sealed class PrivateImplementationDetails : DefaultTypeDef, Cci.INamesp
 
         // field types for different block sizes.
         private ImmutableArray<Cci.ITypeReference> _orderedProxyTypes;
-        private readonly ConcurrentDictionary<uint, Cci.ITypeReference> _proxyTypes = new ConcurrentDictionary<uint, Cci.ITypeReference>();
+        private readonly ConcurrentDictionary<(uint Size, ushort Alignment), Cci.ITypeReference> _proxyTypes = new ConcurrentDictionary<(uint Size, ushort Alignment), Cci.ITypeReference>();
```

Comment:
@jcouv, note that an alternative to most of the changes in this commit would be always setting pack to 8 for anything that's not 1/2/4/8 bytes in length.  If you'd prefer that route, let me know.

Reply:
Also, it wasn't clear to me how the compiler is avoiding bugs when the hash codes happen to collide.  Is the assumption that it simply won't ever happen because it's using a cryptographic hash and the chances of collision are super small?  If there's other logic for handling such collisions, I can also change the naming logic to just piggyback on that instead.